### DISCUSSION TITLE
How to load a .node file (Node-API module)?

### DISCUSSION THREAD

**Discussion by abflow** [denoland/deno#18476] [Q&A] [answered]
Hi,

I am trying to load a local native addon built with Node-API (N-API) which deno supports (https://github.com/denoland/deno/pull/17553).

Firstly, I can not load a local node module with the native addon, because deno does not seem to support the `file:` specifier yet, which is required by package.json for dealing with local node modules (https://github.com/denoland/deno/issues/18474).

Secondly, when I try to load the `.node` file directly into a deno script, it fails to do so with ES module import. 

As far as I understand, a native addon should be loaded with the `require` function(https://github.com/denoland/deno/issues/17246), but this function is not defined in a deno script.

**Comment by aapoalas** [comment, marked as answer]
I just checked that it I at least get some errors related to actual Node Add-on loading with this:
```ts
import { createRequire } from "node:module";

const require = createRequire(import.meta.url);
require("./some_addon.node");
```

**Reply by abflow** [reply]
Thanks @aapoalas for pointing the right direction

At least now, I have the following error, which indicates that my native module can not run yet in Deno because of a N-API protocol not yet supported:

`"napi_add_finalizer is not yet supported."`

**Reply by aapoalas** [reply]
Found the issue for this: https://github.com/denoland/deno/issues/17325
